src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -399,6 +399,28 @@ class ConcatExpression : public detail::VariadicExpression {
 using EncodeForUriExpression =
     StringExpressionImpl<1, decltype(encodeForUriImpl)>;
 
+// TO_NUMERIC (int or double)
+auto InvalidArg = std::errc::invalid_argument;
+auto OutOfRange = std::errc::result_out_of_range;
+
+template <typename T>
+[[maybe_used]] auto toNumeric = [] (std::optional<std::string> input) {
+  if (!input.has_value()) {
+    return Id::makeUndefined();
+  } else {
+    T res{};
+    auto str = boost::lexical_cast<std::string>(*input);
```

Comment:
```suggestion
    const auto& str = input.value();
```
No need for `boost` nor the string copy.